vkcs/compute/resource_instance.go

```diff
@@ -1545,8 +1549,16 @@ func resourceInstancePersonalityV2(d *schema.ResourceData) servers.Personality {
 
 // suppressAvailabilityZoneDetailDiffs will suppress diffs when a user specifies an
 // availability zone in the format of `az:host:node` and Nova/Compute responds with
-// only `az`.
-func suppressAvailabilityZoneDetailDiffs(_, old, new string, _ *schema.ResourceData) bool {
+// only `az`. And suppress diff when instance has power_state `shelved_offloaded`.
+func suppressAvailabilityZoneDetailDiffs(_, old, new string, d *schema.ResourceData) bool {
+	if d.Get("power_state").(string) == "shelved_offloaded" {
```

Comment:
This logic should be placed under resource's plan customization to make `availability_zone` conditionally "requires new".